### ISSUE TITLE
[Action Bar] Add `aria-collapsed` to expand/collapse button

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#7003]
### Summary

Add context to the `action-bar`'s expand/collapse button when it is expanded. Surfaced as part of MapViewer's a11y audit in the Spring of 2022.

### Actual Behavior

Context is provided if the action can be collapsed or expanded. However, no context is provided for AT to know an expansion or collapse has taken place.

### Expected Behavior

Announce if the component is expanded or collapsed to AT by adding an `aria-expanded` property (`true`/`false`) boolean, which provides additional context to AT.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/action-bar/#sample

### Reproduction Steps

1. Open a screen reader of choice (VoiceOver, NVDA, or JAWS)
2. Open the [Developer site sample](https://developers.arcgis.com/calcite-design-system/components/action-bar/#sample)
3. Navigate via keyboard to the "Expand/Collapse" action
4. Observe while the context is provided if the action can be collapsed or expanded, no context is provided for AT to know an expansion or collapse has taken place

### Reproduction Version

1.4.0

### Priority impact

p4 - not time sensitive

### Esri team

Calcite (dev)

**Comment by driskull** [comment]
@geospatialem one issue I'm seeing here is that calcite-action already has a `active` property which translates to `aria-pressed`. It seems like both `aria-pressed` and `aria-expanded` are conflicting with each other. 

Should we maybe refactor this so they can't both be set? Is there some workaround for when the following happens?

```html
<button aria-expanded="true" aria-pressed="true">
  Toggle
</button>
```

**Comment by geospatialem** [comment]
> @geospatialem one issue I'm seeing here is that calcite-action already has a `active` property which translates to `aria-pressed`. It seems like both `aria-pressed` and `aria-expanded` are conflicting with each other.
> 
> Should we maybe refactor this so they can't both be set? Is there some workaround for when the following happens?
> 

That sounds like a good plan forward.

It looks like `aria-pressed` is associated with the `active` property, does that sound right? 

https://github.com/Esri/calcite-design-system/blob/f199b3cafc40438c32a12cad2aef4d26be4526f6/packages/calcite-components/src/components/action/action.tsx#L318

If so, @driskull - what would you think about removing [`aria-pressed`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed), which shouldn't indicate a "selected" or "active" state, but rather adds "press" context for AT users. And instead consider a new translation to the button's `aria-label` indicating its active state, similar to what is added when `indicator` is true?

https://github.com/Esri/calcite-design-system/blob/f199b3cafc40438c32a12cad2aef4d26be4526f6/packages/calcite-components/src/components/action/action.tsx#L302

**Comment by driskull** [comment]
> It looks like aria-pressed is associated with the active property, does that sound right?

Yes, correct.

> what would you think about removing [aria-pressed](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed), which shouldn't indicate a "selected" or "active" state, but rather adds "press" context for AT users. And instead consider a new translation to the button's aria-label indicating its active state, similar to what is added when indicator is true?

Shouldn't we keep aria-pressed because some of these buttons are toggle buttons. right? `active` means that its "on". If we remove it, we lose that context. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed

**Comment by driskull** [comment]
I think we'll need to setup some time to discuss a plan moving forward here.

We already have a property for toggling (active) but we don't have a property for expand status. We would need a new property to handle this because `aria-expanded` needs to be added to the internal button element.

**Comment by geospatialem** [comment]
> We already have a property for toggling (active) but we don't have a property for expand status. We would need a new property to handle this because `aria-expanded` needs to be added to the internal button element.

Ah, gotcha, thanks for the context - was initially thinking the `action` component was providing a selection state, rather than leveraging `aria-pressed` - it's more clear now.

Concur with your [original statement above](https://github.com/Esri/calcite-design-system/issues/7003#issuecomment-1870697923)- we'd want to ensure `aria-pressed` and `aria-expanded` aren't used together with the `action-bar`, and when the component is expanded, ensure `aria-expanded` takes precedence. Per your proposal above, a new property with conditions sounds like the best path forward, if you still agree?

**Comment by driskull** [comment]
Yes, I agree. We'll have to deprecate `active` and have a new property for handling toggled/expanded state.

**Comment by driskull** [comment]
I guess we would have to add something like:

```tsx
state: "pressed" | "unpressed" | "expanded" | "collapsed";
```

```tsx
state: "on" | "off" | "expanded" | "collapsed";
```

Would we need to support the `mixed` state of aria-pressed?

**Comment by driskull** [comment]
When something is "expanded" would it get the same background color affordance as when "pressed/active"?

**Comment by driskull** [comment]
I guess we could just add an `expanded` property on action which would take precedence over `active`? What do you think? is that easier than a single mode?

**Comment by driskull** [comment]
I think this needs a design/api discussion before it can move forward.

**Comment by geospatialem** [comment]
The above is blocked and sending to the freezer 🥶 until accessibility API supports shadow DOM further for implementation. As this is not currently feasible within shadow DOM, since the component relies on an internal `button` via the shadow DOM.

Some resources for consideration and checking on the Accessibility object model (AOM), once supported:
- AOM: https://wicg.github.io/aom/explainer.html#the-accessibility-object-model
- Cross root ARIA delegation proposal: https://github.com/leobalter/cross-root-aria-delegation/blob/main/explainer.md
- Great overview post: https://nolanlawson.com/2022/11/28/shadow-dom-and-accessibility-the-trouble-with-aria/

**Comment by driskull** [comment]
If we add a property of `type` to `calcite-action` of `type: "button" | "toggle" | "expand"` that would get us closer.

We could use `active` for toggle and expand or we could add another property just for `expanded`. Thoughts here? @jcfranco @geospatialem

**Comment by driskull** [comment]
The reason we need a different type for expand is that expand uses aria-expanded instead of aria-pressed

**Comment by driskull** [comment]
@geospatialem can you comment on the following:

> We could use active for toggle and expand or we could add another property just for expanded. Thoughts here? @jcfranco @geospatialem 

Would we want to share the `active` property between these two? or separately use `expanded` for expand buttons and `active` for toggle buttons? Should we use `selected` instead of `active` for consistency? @jcfranco

**Comment by driskull** [comment]
@geospatialem do we know if `aria-expanded` is needed on the internal button element or if developers can apply the `aria-expanded` to the host action element? Trying to see if we even need to add a property for `aria-expanded`. Let me know what you think. cc @jcfranco

**Comment by driskull** [comment]
@jcfranco another reason why we may need to have an `expand` type is that internally if the action type of `"toggle"` is set it will set `aria-pressed` internally on the button.  I don't think we can have both `aria-pressed` and `aria-expanded` on a button.  cc @geospatialem 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role#associated_aria_roles_states_and_properties

I'm not sure this would be any better?

```html
<calcite-action aria-expanded="true">
   <button aria-pressed="true" /> // internal shadow button
</calcite-action>
```

**Comment by driskull** [comment]
> The toggle type seems different since it also changes the component’s behavior.

They are essentially the same, the underlying difference is which aria-attribute they use. `aria-expanded` vs `aria-pressed`

> That said, we don’t document this anywhere, so it’s more of a hidden feature. 

I'd also like to know if this is actually working as it should. Essentially, aria-expanded is set on both the host and the internal button. Does this duplication have any issues and cause more harm than good? cc @geospatialem

**Comment by driskull** [comment]
I have a codepen to test out using `ariaControlsElements` within a shadow DOM.

https://codepen.io/driskull/pen/QwbjVXY?editors=1010

**Comment by driskull** [comment]
The aria properties were added to Lumina. @jcfranco can we update to the latest Lumina? https://devtopia.esri.com/WebGIS/arcgis-web-components/pull/6394

**Comment by jcfranco** [comment]
Blocked by https://github.com/Esri/calcite-design-system/pull/12465.

**Comment by isaacbraun** [comment]
Verified on `3.3.0-next.115` via https://codepen.io/isaacbraun/pen/YPwqdvg with VoiceOver, Jaws, and NVDA.
